File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.c


int countMatches(char *** items, int itemsSize, int* itemsColSize, char * ruleKey, char * ruleValue){

    /// There are three series in a series. items[type,color,name];
    
    int count = 0;
    int i=0;
    
    for(i=0; i<itemsSize; i++)
    {
        if(strcmp(ruleKey, "type") == 0)
        {
            if(strcmp(ruleValue, items[i][0]) == 0)
            {
                count++;
            } 
        }
        else if(strcmp(ruleKey, "color") == 0)
        {
            if(strcmp(ruleValue, items[i][1]) == 0)
            {
                count++;
            } 
        }
        else if(strcmp(ruleKey, "name") == 0)
        {
            if(strcmp(ruleValue, items[i][2]) == 0)
            {
                count++;
            }
        }        
          
    }
     return count;   
}

File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.java
class Solution {
    public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
        int count = 0;
        int j = ruleKey.equals("color") ? 1 : ruleKey.equals("name") ? 2 : 0;
        for (int i = 0;i <items.size();i++){
            if (items.get(i).get(j).equals(ruleValue)){
                count += 1;
            }
        }
        return count;
    }
}